test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java

```diff
@@ -80,15 +88,30 @@ public boolean isClientParameter(String name) {
      * Parses the complete set of REST spec available under the provided directories
      */
     public static ClientYamlSuiteRestSpec load(String classpathPrefix) throws Exception {
-        Path dir = PathUtils.get(ClientYamlSuiteRestSpec.class.getResource(classpathPrefix).toURI());
         ClientYamlSuiteRestSpec restSpec = new ClientYamlSuiteRestSpec();
         ClientYamlSuiteRestApiParser restApiParser = new ClientYamlSuiteRestApiParser();
-        try (Stream<Path> stream = Files.walk(dir)) {
-            stream.forEach(item -> {
-                if (item.toString().endsWith(".json")) {
-                    parseSpecFile(restApiParser, item, restSpec);
+        URL specResource = ClientYamlSuiteRestSpec.class.getResource(classpathPrefix);
+
+        if (specResource.getFile().contains(".jar")) {
+            int indexOf = specResource.getFile().indexOf(".jar");
+            String jarFileName = specResource.getFile().substring(0, indexOf + 4);
+            JarFile jarFile = new JarFile(jarFileName);
+            Enumeration<JarEntry> e = jarFile.entries();
+            while (e.hasMoreElements()) {
+                JarEntry entry = e.nextElement();
+                if (entry.getName().startsWith( classpathPrefix) && entry.getName().endsWith(".json")) {
+                    parseSpecFile(restApiParser, null/*entry.getName()?*/, restSpec);
```

Comment:
this side of the branch does not quite work yet, currently stuck at `java.security.AccessControlException: access denied ("java.io.FilePermission" "file:/home/javanna/elastic/elasticsearch/rest-api-spec/build/distributions/rest-api-spec-7.0.0-alpha1-SNAPSHOT.jar" "read")` at line 98 above.